src/MarginCalculator.php

```diff
@@ -67,52 +67,62 @@ protected function getCandidateIdToRankMap(Ballot $ballot) : array
     }
 
     /**
```

Comment:
This approach may prove more testable than the one I initially proposed in this pull request. I'm open to alternatives.